### PR TITLE
refactor: use pathlib instead of os.path

### PR DESCRIPTION
**Pull request by vitorhcl**
pathlib provides a much cleaner spearation between strings and paths, leaves "/" operator for joining paths and also provides some builtin methods, improving the code readabilty and mantainability.

### CODE REVIEW

File: tldr.py [hunk 1]
```diff
@@ -507,8 +508,8 @@ def main() -> None:
         print(get_commands(options.platform))
     elif options.render:
         for command in options.command:
-            if os.path.exists(command):
-                with open(command, encoding='utf-8') as open_file:
+            if Path(command).exists:
```
Comment (JacobCallahan) [inline tldr.py:511]: This is a method call
```suggestion
            if Path(command).exists():
```

**Review by JacobCallahan — CHANGES_REQUESTED** [review]
love pathlib!

**Review by kbdharun — APPROVED** [review]

**Comment by kbdharun** [comment]
Interestingly, this is failing CI for older versions of Python.

File: tests/test_tldr.py [hunk 2]
```diff
@@ -120,17 +122,17 @@ def test_get_platform(platform, expected):
...
     monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
     monkeypatch.setenv("HOME", "/tmp/home")
-    assert tldr.get_cache_dir() == "/tmp/home/.cache/tldr"
+    assert tldr.get_cache_dir() == Path("/tmp/home/.cache/tldr")
 
 
 def test_get_cache_dir_default(monkeypatch):
     monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
     monkeypatch.delenv("HOME", raising=False)
     monkeypatch.setattr(os.path, 'expanduser', lambda _: '/tmp/expanduser')
```
Comment (vitorhcl) [inline tests/test_tldr.py:137]:
```suggestion
     monkeypatch.setattr(pathlib.Path, 'home', lambda _: '/tmp/expanduser')
```

**Review by owenvoke — APPROVED** [review]

**Pull request merged**